src/vehicle.cpp

```diff
@@ -1557,6 +1564,22 @@ void VehicleEnterDepot(Vehicle *v)
 }
 
 
+/**
+* Move to the x, y coordinates as part of regular travel.
+* @param x  new x position
+* @param y  new y position
+*/
+void Vehicle::Travel(int x, int y)
+{
+	/* Update cargotiles delta distance */
+	int delta = abs(this->x_pos - x) + abs(this->y_pos - y);
+	this->cd_distance += delta;
```

Comment:
Not quite sure how this doesn't massively overflow. `cd_distance` is a byte and `Travel` is called every time the vehicle moves. The x/y doesn't specify whether it's tiles or subtiles, even if it's tiles, any journey of 255 tiles would overflow? If it's subtiles, that would happen even more regularly? What am I missing?

Reply:
It's subtile position, and yes `cd_distance` probably should be widened.